components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/handler/IndirectTokenRevocationHandler.java

```diff
@@ -0,0 +1,15 @@
+package org.wso2.carbon.identity.oauth.handler;
+
+import org.wso2.carbon.user.core.UserStoreException;
+import org.wso2.carbon.user.core.UserStoreManager;
+
+/**
+ * Token persistence handler interface used to abstract the token persistence logic.
+ * {@link DefaultIndirectTokenRevocationHandlerImpl} is used to preserve the existing token persistence behaviour.
+ * An extension can be used to remove or modify the token persistence behaviour.
+ */
+public interface IndirectTokenRevocationHandler {
```

Comment:
The interface and the purpose of the interface explained in the comment is not clear. Instead of mentioning what is existing (which can change at any point) let's explain what this interface is designed for.

Reply:
Also instead of the word indirect let's use implicit

Reply:
What is the difference between this interface and OAuth2RevocationProcessor? shouldn't handling implicit revocation should also be a responsibility of OAuth2RevocationProcessor?

Reply:
OAuth2RevocationProcessor was added to handle the direct token revocation flow. But as you have pointed out, the same can be used to handle the revocation responsibility in general. Hence +1 for using the same interface.

Reply:
moved to OAuth2RevocationProcessor with https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2179/commits/69f8c55fb76f15cbfe22b3c551b17491dc1796b0